src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Utils.java

```diff
@@ -27,46 +27,27 @@
 package jdk.internal.clang;
 
 import jdk.incubator.foreign.CSupport;
+import jdk.incubator.foreign.MemoryAccess;
 import jdk.incubator.foreign.MemoryAddress;
-import jdk.incubator.foreign.MemoryHandles;
 import jdk.incubator.foreign.MemoryLayout;
 import jdk.incubator.foreign.MemorySegment;
-import java.lang.invoke.VarHandle;
 
 public class Utils {
-    public static final VarHandle BYTE_VH = CSupport.C_CHAR.varHandle(byte.class);
-    public static final VarHandle BYTE_ARR_VH = MemoryHandles.withStride(BYTE_VH, 1);
-    public static final VarHandle INT_VH = CSupport.C_INT.varHandle(int.class);
-    public static final VarHandle LONG_VH = CSupport.C_LONGLONG.varHandle(long.class);
-    public static final VarHandle POINTER_VH = MemoryHandles.asAddressVarHandle(CSupport.C_POINTER.varHandle(long.class));
-    public static final VarHandle POINTER_ARR_VH = MemoryHandles.withStride(POINTER_VH, 8);
 
-    static int getInt(MemoryAddress addr) {
-        return (int)INT_VH.get(addr);
+    static int getInt(MemorySegment addr) {
+        return MemoryAccess.getInt(addr);
     }
 
-    static void setInt(MemoryAddress addr, int i) {
-        INT_VH.set(addr, i);
+    static void setLong(MemorySegment addr, long i) {
+        MemoryAccess.setLong(addr, i);
     }
 
-    static int getLong(MemoryAddress addr) {
-        return (int)LONG_VH.get(addr);
```

Comment:
Do we need this renaming wrapper? BTW most of this Utils class is just forwarding to MemoryAccess. Might as well use MemoryAccess directly in callsties?

Reply:
We could, yes